dcc-portal-ui/app/scripts/pathwayviewer/js/directive.js

```diff
@@ -342,7 +334,16 @@
             return;
           }else if(!rendered){
             $('.pathwaysvg').remove();
-            controller.render(xml,zoomedOn);
+            
+            controllerSettings.model = new PathwayModel(xml);
+            controller = new ReactomePathway(controllerSettings);
+            controller.render(zoomedOn);
+
+            var legendSVGElement = controller.getLegendSVGElement(270, 671);
+            d3.select('.pathway-legend-content').append(function(){
+              return legendSVGElement;
```

Comment:
I actually tried that and it threw an error -- I think the append method expects a string or function according to the D3 API https://github.com/d3/d3-3.x-api-reference/blob/master/Selections.md#append